packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,426 @@
+import path from 'path';
+import chalk from 'chalk';
+import { promises as fs } from 'fs';
+import { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem, TypeMembers, StructMember } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const EXPORT_FILE = 'openzeppelin-cli-export.json';
+const PROJECT_FILE = path.join(OPEN_ZEPPELIN_FOLDER, 'project.json');
+const SUCESS_CHECK = chalk.keyword('green')('✔') + ' ';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const manifestsMigrationData = await migrateManifestFiles();
+
+  const { compiler } = await getProjectFile();
+  const exportData = {
+    networkFiles: manifestsMigrationData,
+    compiler,
+  };
+
+  await writeJSONFile(EXPORT_FILE, exportData);
+  console.log(SUCESS_CHECK + `Migration data exported to ${EXPORT_FILE}`);
+  await deleteLegacyFiles();
+
+  console.log("\nThese were your project's compiler options:");
+  console.log(compiler);
+}
+
+async function migrateManifestFiles() {
+  const oldManifests = await getManifests();
+  const migratableManifests = oldManifests.filter(manifest => !isDevelopmentNetwork(getNetworkName(manifest)));
+  const { newManifests, manifestsMigrationData } = await migrateManifests(migratableManifests);
+
+  for (const manifestFile in newManifests) {
+    const newManifest = newManifests[manifestFile];
+    const newFileName = getNewManifestLocation(manifestFile);
+    await writeJSONFile(newFileName, newManifest);
+    console.log(SUCESS_CHECK + `Successfully migrated ${newFileName}`);
+  }
+
+  return manifestsMigrationData;
+}
+
+async function deleteLegacyFiles(): Promise<void> {
+  const oldManifests = await getManifests();
```

Comment:
Can we make it so `getManifests` is only called once?

Reply:
We could, I did it this way because I thought making it a parameter of this function might be confusing/obscure since it's deleting more than legacy manifests.

Reply:
I think it's best to read it once only.